controllers/favorites.js

```diff
@@ -0,0 +1,82 @@
+
+const { response } = require('express');
+const db = require('../db');
+const knex = require('knex')({
+    client: 'pg',
+    connection: {
+        database: 'citymeal',
+        user: process.env.PGUSERNAME,
+        password: process.env.PGPASSWORD
+    }
+});
+
+class Favorites {
+    
+    async saveFavorite(req, res) {
+        const { user_id } = req.params
```

Comment:
You don't want to get user_id from params because that allows user A to manipulate data for another user. It's best if you get the user id from the verification token by using req.user.id!